pip/req/req_set.py

```diff
@@ -200,9 +200,7 @@ def add_requirement(self, install_req, parent_req_name=None):
             requirement is applicable and has just been added.
         """
         name = install_req.name
-        if ((not name or not self.has_requirement(name)) and not
-                install_req.match_markers()):
-            # Only log if we haven't already got install_req from somewhere.
+        if not install_req.match_markers():
```

Comment:
Ah, markers are a property of the dependency relation, not the req itself? Ok, this change makes sense.